service/src/main/java/uk/gov/hmcts/reform/fpl/service/RepresentativeService.java

```diff
@@ -215,28 +214,27 @@ private void associatedRepresentativeWithParty(CaseData caseData, Element<Repres
         }
     }
 
-    public List<Element<Representative>> getRepresentativePartiesToNotify(CallbackRequest request) {
-        CaseData caseData = mapper.convertValue(request.getCaseDetails().getData(), CaseData.class);
-        CaseData caseDataBefore = mapper.convertValue(request.getCaseDetailsBefore().getData(), CaseData.class);
-
-        if (isNotEmpty(caseDataBefore.getRepresentatives())) {
-            if (!caseDataBefore.getRepresentatives().containsAll(caseData.getRepresentatives())) {
-                return getChangedRepresentatives(caseData, caseDataBefore);
+    public List<Element<Representative>> getRepresentativePartiesToNotify(List<Element<Representative>>
+                                                                              currentRepresentatives,
+        List<Element<Representative>> representativesBefore) {
+        if (isNotEmpty(representativesBefore)) {
+            if (!representativesBefore.containsAll(currentRepresentatives)) {
+                return getChangedRepresentatives(currentRepresentatives, representativesBefore);
             }
         } else {
-            if (!caseData.getRepresentatives().isEmpty()) {
-                return caseData.getRepresentatives();
+            if (!currentRepresentatives.isEmpty()) {
```

Comment:
Do you think another `isNotEmpty` would be better than negating?

Reply:
agree :)